helios-client/src/main/java/com/spotify/helios/client/DefaultRequestDispatcher.java

```diff
@@ -77,20 +73,16 @@
 
   private static final Logger log = LoggerFactory.getLogger(DefaultRequestDispatcher.class);
 
-  private static final long RETRY_TIMEOUT_MILLIS = SECONDS.toMillis(60);
   private static final long HTTP_TIMEOUT_MILLIS = SECONDS.toMillis(10);
-  private static final List<String> VALID_PROTOCOLS = ImmutableList.of("http", "https");
-  private static final String VALID_PROTOCOLS_STR =
-      String.format("[%s]", Joiner.on("|").join(VALID_PROTOCOLS));
 
-  private final Supplier<List<URI>> endpointSupplier;
+  private final Iterator<Endpoint> endpointIterator;
```

Comment:
Ah I see that the `EndpointIterator` will cycle over the elements. In that case I'd declare this field as `EndpointIterator` to make that behavior clear, as this class depends on that behavior as opposed to the Iterator behavior. 

(It'd be even nicer if there was some interface to use in the declaration like `CyclingIterator`)